waspc/e2e-tests/Main.hs

```diff
@@ -15,13 +29,55 @@ main = do
 
 -- TODO: Investigate automatically discovering the tests.
```

Comment:
I think it might be worth it to investigate `tasty-discover` in this PR, as we're adding a way longer list of tests.